### ISSUE TITLE
Catboost support

### ISSUE THREAD

**Issue by rspadim** [dmlc/treelite#83]

**Comment by hcho3** [comment]
@rspadim Are you familiar with how Catboot saves models?

**Comment by rspadim** [comment]
there're some types, and very interesting since some are wellknown formats (standards)
1) Onnx (open wellknown format, but don't support features yet)
2) CatboostBinary
3) AppleCoreML
4) Json
5) "default"

i will point it here, and try to upload some samples
https://github.com/catboost/catboost/blob/master/catboost/libs/model/model.cpp#L174

**Comment by rspadim** [comment]
well... now i know that it can export to python and c++ too

https://github.com/catboost/tutorials/blob/7897e65888889f559715bf0da1222a9b6dbc9d6b/apply_model/model_export_as_cpp_code_tutorial.md

https://github.com/catboost/tutorials/blob/7897e65888889f559715bf0da1222a9b6dbc9d6b/apply_model/model_export_as_python_code_tutorial.md

sorry my fault

**Comment by hcho3** [comment]
@rspadim It may be still beneficial to add Catboost support in Treelite, if you'd like to generate pure C core without external dependency. That said, closing for now.

**Comment by rspadim** [comment]
an exampel to create models:


import logging
from sklearn.datasets import load_boston
X, y = load_boston(return_X_y=True)
print('dimensions of X = {}'.format(X.shape))
print('dimensions of y = {}'.format(y.shape))
#import xgboost
#dtrain = xgboost.DMatrix(X, label=y)
#params = {'max_depth':3, 'eta':1, 'silent':1, 'objective':'reg:linear',
#          'eval_metric':'rmse'}
#bst = xgboost.train(params, dtrain, 20, [(dtrain, 'train')])

from catboost import CatBoostClassifier, CatBoostRegressor
model = CatBoostRegressor(
#    custom_loss=['Accuracy'],
#    random_seed=42,
#    logging_level='Silent'
)
model.fit(X, y)
for i in ['CPP', 'python', 'json', 'bin', 'onnx', 'applecoreml']:
    print(i)
    model.save_model("teste.catboost."+ i, i)

**Comment by rspadim** [comment]
hi @hcho3 , yes i agree with you, pure C is the best think treelite have, it's very intuitive to port to any language

**Comment by hcho3** [comment]
@rspadim It is the job of Treelite contributors to support Catboost, not Catboost contributors. I think you should close catboost/catboost#721.

Also re-opening this issue, as a reminder that we should support Catboost

**Comment by rspadim** [comment]
Nice, reference to create treelite support

https://github.com/catboost/catboost/blob/6ec9193d0cbb9a8712bd8fe434b823f30c4ee875/catboost/libs/model/model.cpp#L164

cpp might be easy to port

**Comment by hcho3** [comment]
Closing in favor of #377
